server/routes/removeGoal/RemoveGoalController.test.ts

```diff
@@ -0,0 +1,78 @@
+import { NextFunction, Request, Response } from 'express'
+import handoverData from '../../testutils/data/handoverData'
+import RemoveGoalController from './RemoveGoalController'
+import GoalService from '../../services/sentence-plan/goalService'
+import mockReq from '../../testutils/preMadeMocks/mockReq'
+import mockRes from '../../testutils/preMadeMocks/mockRes'
+import { testGoal } from '../../testutils/data/goalData'
+import locale from './locale.json'
+import URLs from '../URLs'
+
+jest.mock('../../services/sessionService', () => {
+  return jest.fn().mockImplementation(() => ({
+    getSubjectDetails: jest.fn().mockReturnValue(handoverData.subject),
+  }))
+})
+
+jest.mock('../../services/sentence-plan/goalService', () => {
+  return jest.fn().mockImplementation(() => ({
+    removeGoal: jest.fn().mockReturnValue({ status: 204 }),
+    getGoal: jest.fn().mockReturnValue(testGoal),
+    updateGoal: jest.fn().mockReturnValue(testGoal),
+  }))
+})
+
+describe('RemoveGoalController', () => {
+  let controller: RemoveGoalController
+  let mockGoalService: jest.Mocked<GoalService>
+  let req: Request
+  let res: Response
+  let next: NextFunction
+  const viewData = {
+    data: {
+      popData: handoverData.subject,
+      type: 'some-type',
+      goal: testGoal,
+    },
+    errors: {},
+    locale: locale.en,
+  }
+
+  beforeEach(() => {
+    mockGoalService = new GoalService(null) as jest.Mocked<GoalService>
+    req = mockReq()
+    res = mockRes()
+    next = jest.fn()
+    req.query.type = 'some-type'
+    controller = new RemoveGoalController(mockGoalService)
+  })
+
+  describe('get', () => {
+    it('should render without validation errors', async () => {
+      await controller.get(req, res, next)
+      expect(res.render).toHaveBeenCalledWith('pages/remove-goal', viewData)
+    })
+  })
+
+  describe('post', () => {
+    it('should return to plan-summary if cancel removal', async () => {
+      req = { body: { type: 'some-type', action: 'cancelRemove' } } as Request
+      await controller.post(req as Request, res as Response, next)
+      expect(res.redirect).toHaveBeenCalledWith(`${URLs.PLAN_SUMMARY}?type=some-type`)
+    })
+
+    it('should return to plan-summary with removing goal if cancel removal', async () => {
```

Comment:
isn't this the same test as above but with an additional assert?

Reply:
Yeah didnt realized that, removed one.

Reply:
should the title in the `it` line be "with*out* removing goal" ?

Reply:
> should the title in the `it` line be "with_out_ removing goal" ?
yes its updated now